### PR TITLE
Add docstring for Dates.adjust

### PR DESCRIPTION
**Pull request by jam-khan**
This is a PR for the issue #52725.

Thanks!

### CODE REVIEW

**Comment by jariji** [comment]
I'm not actually sure this function is supposed to be public. https://github.com/JuliaLang/julia/issues/52746

**Comment by stevengj** [comment]
You also need to update the docstring test for the Dates module, similar to https://github.com/JuliaLang/julia/pull/52913#issuecomment-1892885864

**Comment by stevengj** [comment]
Regardless of whether it should be exported/public or not, it doesn't hurt to have a docstring.

File: stdlib/Dates/src/adjusters.jl [hunk 1]
```diff
@@ -196,6 +196,28 @@ end
 Base.show(io::IO, df::DateFunction) = println(io, df.f)
 
 # Core adjuster
+
+"""
+    adjust(df::DateFunction, start, step, limit) -> TimeType
+
+Adjusts the date in `start` until `f::Function` in `df::DateFunction` returns `true`.
+The step size dictates change in `start` on every iteration. If `limit` iterations occur,
+then ArgumentError is thrown.
```
Comment (stevengj) [inline stdlib/Dates/src/adjusters.jl:205]:
```suggestion
then an [`ArgumentError`] is thrown.
```

File: stdlib/Dates/src/adjusters.jl [hunk 2]
```diff
@@ -204,6 +226,27 @@ function adjust(df::DateFunction, start, step, limit)
     throw(ArgumentError("Adjustment limit reached: $limit iterations"))
 end
 
+"""
+    adjust(df::DateFunction, start) -> TimeType
```
Comment (stevengj) [inline stdlib/Dates/src/adjusters.jl:230]: Better to have a single docstring

File: stdlib/Dates/src/adjusters.jl [hunk 3]
```diff
@@ -196,6 +196,28 @@ end
...
+
+julia> adjust(date -> year(date) == 2025, Date(2022, 1, 1), step=Year(1), limit=4)
+2025-01-01
+
+julia> adjust(date -> day(date) == 15, Date(2022, 1, 1); step=Year(1), limit=3)
+ERROR: ArgumentError: Adjustment limit reached: 3 iterations
+Stacktrace:
+[...]
+```
+"""
```
Comment (stevengj) [inline stdlib/Dates/src/adjusters.jl:220]:
```suggestion
```
Only document the `Function` interface below, since `DateFunction` is private.

**Review by LilithHafner — APPROVED** [review]

**Comment by vtjnash** [comment]
It appears that either this or https://github.com/JuliaLang/julia/pull/53027 needs to be reverted, due to a merge conflict between them that is breaking CI. Which is better to revert and reland?

**Comment by LilithHafner** [comment]
The CI on master after they both merged looks green-ish to me:

https://buildkite.com/julialang/julia-master/builds/32740

**Comment by giordano** [comment]
`doctest` job is very much red in the build you linked.

**Comment by LilithHafner** [comment]
Thanks! I'm so used to red that I instinctively ignored the fact that a single check failed.

**Pull request merged**